src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzerTests.cs

```diff
@@ -0,0 +1,694 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.PlatformCompatabilityAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices.UnitTests
+{
+    public partial class PlatformCompatabilityAnalyzerTests
+    {
+        [Fact]
+        public async Task OsDependentMethodsCalledWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    public void M1()
+    {
+        [|WindowsOnly()|];
+        [|Obsoleted()|];
+        [|Removed()|];
+    }
+    [MinimumOSPlatform(""Windows10.1.1.1"")]
+    public void WindowsOnly()
+    {
+    }
+    [ObsoletedInOSPlatform(""Linux4.1"")]
+    public void Obsoleted()
+    {
+    }
+    [RemovedInOSPlatform(""Linux4.1"")]
+    public void Removed()
+    {
+    }
+}
+" + MockAttributesSource;
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task WrongPlatformStringsShouldHandledGracefully()
```

Comment:
I'm trying to understand this test: Why do invalid strings still trigger the rule?

Reply:
Good question, the purpose of this analyzer is not reporting the wrong string values, the purpose of this test is to check if the analyzer would crush or throw any unhandled exceptions with any different/wrong OS platform string values